sbndqm/dqmAnalysis/CRT/SBND/BernCRTdqmSBND_module.cc

```diff
@@ -339,182 +327,365 @@ if(!fragmentHandle.isValid() || fragmentHandle->size() == 0)
-	 *    Average: Every reporting_interval, the average of all metric values since the last report is sent (for rates)
+
+    if(isTs0Reset)
+      {
+	++nT0Resets[mac5];
+
+	if(ts0Good)
+	  t0Reset[mac5] = ts0;
+      }
+
+    if(isTs1Reset)
+      {
+	++nT1Resets[mac5];
 	
-    * We note that the level here is used as a threshold; we send any value of the metric above the level. For the CRT DQM, all levels = 0.
-    */
-
-    sbndaq::sendMetric("CRT_board", readout_number_str, "MaxADCValue", max, 0, artdaq::MetricMode::LastPoint);
-    sbndaq::sendMetric("CRT_board", readout_number_str, "MaxADCChannel", maxindex + 32 * mac5, 0, artdaq::MetricMode::LastPoint);
-    sbndaq::sendMetric("CRT_board", readout_number_str, "Flag", thisflag, 0, artdaq::MetricMode::LastPoint);
-    sbndaq::sendMetric("CRT_board", readout_number_str, "baseline", baseline, 0, artdaq::MetricMode::Average);
-    sbndaq::sendMetric("CRT_board", readout_number_str, "TS0", ts0, 0, artdaq::MetricMode::LastPoint);
-    sbndaq::sendMetric("CRT_board", readout_number_str, "TS1", ts1, 0, artdaq::MetricMode::LastPoint);
-    sbndaq::sendMetric("CRT_board", readout_number_str, "Deadtime", deadtime, 0, artdaq::MetricMode::Minimum);
-    sbndaq::sendMetric("CRT_board", readout_number_str, "MissingT0", missingT0, 0, artdaq::MetricMode::Maximum);
-    sbndaq::sendMetric("CRT_board", readout_number_str, "MissingT1", missingT1, 0, artdaq::MetricMode::Maximum);
+	if(ts0Good)
+	  {
+	    if(t1Reset[mac5] != std::numeric_limits<uint32_t>::max())
+	      {
+		uint32_t fracTDCT1Reset = tdcT1Reset % static_cast<uint32_t>(1e9);
+		uint32_t diff           = ts0 > fracTDCT1Reset ? ts0 - fracTDCT1Reset : fracTDCT1Reset - ts0;
+		uint32_t currDiff       = t1Reset[mac5] > fracTDCT1Reset ? t1Reset[mac5] - fracTDCT1Reset : fracTDCT1Reset - t1Reset[mac5];
+		
+		if(diff < currDiff)
+		  t1Reset[mac5] = ts0;
+	      }
+	    else
+	      t1Reset[mac5] = ts0;
+	  }
+      }    
+
+    // Documentation on sbndaq::sendMetric lives in MetricManager.hh
+
+    sbndaq::sendMetric("CRT_board", mac5_str, "MissingT0", missingT0[mac5], 0, artdaq::MetricMode::Maximum);
+    sbndaq::sendMetric("CRT_board", mac5_str, "MissingT1", missingT1[mac5], 0, artdaq::MetricMode::Maximum);
  
     //only send clockdrift info when it makes sense to do so; that is, for T0 reset events.
-    if(isTS0 && isTS0good) {sbndaq::sendMetric("CRT_board", readout_number_str, "T0clockdrift", ts0 - 1e9, 0, artdaq::MetricMode::LastPoint);}
-    if(isTS1 && isTS1good) {sbndaq::sendMetric("CRT_board", readout_number_str, "T1clockdrift", ts1 - 1e9, 0, artdaq::MetricMode::LastPoint);}
+    if(isTS0 && isTS0good) {sbndaq::sendMetric("CRT_board", mac5_str, "T0clockdrift", static_cast<int>(ts0) - 1e9, 0, artdaq::MetricMode::LastPoint);}
+    //if(isTS1 && isTS1good) {sbndaq::sendMetric("CRT_board", mac5_str, "T1clockdrift", ts1 - 1e9, 0, artdaq::MetricMode::LastPoint);}
```

Comment:
remove

Reply:
Removed for now but if we do implement a T1 clockdrift we'll revisit this

Reply:
Yeah - the `T1ResetTDCDiff` is essentially the T1 drift metric. Unlike T0 it isn't a set phase clock so the implementation is more complex & requires the TDC values as the reference point.